app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -214,6 +214,8 @@ class NavigationOverlayFragment : Fragment() {
             requireWebRenderComponents.sessionManager,
             requireWebRenderComponents.tabsUseCases,
             ::closeOverlay)
+
+        pocketViewModel.state.subscribe(tabsTray.pocketStateObserver)
```

Comment:
You should see a lint error here that says "the result of subscribe is not used."  That's reminding you that we need to handle the `Disposable` here, and clean up the subscription after the view has been destroyed.  The easiest way to do that would be to move this into `onStart`, and use the RxKotlin `addTo` method to add to an existing `CompositeDisposable`.

```kotlin
pocketViewModel.state.subscribe(tabsTray.pocketStateObserver)
    .addTo(compositeDisposable)
```

A `CompositeDisposable` is basically just a list of `Disposables` that lets you clean all of them up at once.  You can look at how `onStart` and `onStop` use `observePocketState` for an example usage.